sweepai/utils/safe_dictionary.py

```diff
@@ -0,0 +1,52 @@
+from threading import Lock
+
+
+class SafeDictionary:
+    def __init__(self):
+        self._dict = {}
+        self._lock = Lock()
+
+    def get(self, key, default=None):
+        with self._lock:
+            return self._dict.get(key, default)
+
+    def set(self, key, value):
+        with self._lock:
+            self._dict[key] = value
+
+    def delete(self, key):
+        with self._lock:
+            if key in self._dict:
+                del self._dict[key]
+
+    def contains(self, key):
+        with self._lock:
```

Comment:
**Enhance SafeDictionary class with proper type hinting for better code clarity and static type checking.** 
 
 Improve type hinting in the SafeDictionary class to enhance code clarity and enable better static type checking. 
```suggestion
from threading import Lock
from typing import Any, Dict, Optional, TypeVar

K = TypeVar('K')
V = TypeVar('V')

class SafeDictionary(Dict[K, V]):
    def __init__(self):
        self._dict: Dict[K, V] = {}
        self._lock: Lock = Lock()

    def get(self, key: K, default: Optional[V] = None) -> Optional[V]:
        with self._lock:
            return self._dict.get(key, default)

    def set(self, key: K, value: V) -> None:
        with self._lock:
            self._dict[key] = value

    def delete(self, key: K) -> None:
        with self._lock:
            if key in self._dict:
                del self._dict[key]

    def contains(self, key: K) -> bool:
        with self._lock:
            return key in self._dict
```